model_checking/src/symbolic.rs

```diff
@@ -121,8 +121,9 @@ pub fn new_ctx() -> VirtContext {
     // We fix the architecture type to 64 bits
     ctx.csr.misa = (0b10 << 62) | (ctx.csr.misa & ((1 << 62) - 1));
 
-    // We must have support for usermode in Miralis
+    // We must have support for usermode and in Miralis
```

Comment:
and ? Typo here^^

Is there an issue if there is no S mode? Miralis should support both with and without S mode (but he hardware capabilities must be updated accordingly I think?)

Reply:
Currently the interrupt virtualisation process won't work if we don't have support for S-mode. Let me handle this in a second PR. The core of the logic is there and can be merged.